### PR TITLE
[build] Remove leftovers of code signing / OSX IDE infrastructure

### LINKED ISSUE #7476: Codesigning in `Makefile`
There is some special code in the makefile to sign OSX binaries [see `LINKMETADATA` et friends]

However, @maximedenes mentioned that the signing does happen today at the package level, so I wonder what's going on. Is that dead code?

There is no real documentation for this feature other than 9b0dbb30d653bf35145e1e77f0fecfc9d79aa81d .

I have opened PR #7475 to see how the packages look now.

### PR DESCRIPTION
**Pull request by ejgallego**
This is not used anymore, and after #14122 the makefile parts for the
dmg generation are not used anymore.

Closes #7476 .

### CODE REVIEW

File: doc/changelog/11-infrastructure-and-dependencies/13617-make+use_dune_for_ocaml.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- **Changed:**
+  Coq's OCaml parts and tools [coq-core] are now built using Dune.
```
Comment (jfehrle) [inline doc/changelog/11-infrastructure-and-dependencies/13617-make+use_dune_for_ocaml.rst:2]:
```suggestion
  Coq's OCaml code and tools [coq-core] are now built using Dune.
```

**Comment by Zimmi48** [comment]
I confirm that the internal signing infrastructure is not used anymore.

**Comment by jfehrle** [comment]
"codesigning" is too easily misread as "co-designing" rather than "code signing".

**Review by Zimmi48 — APPROVED** [review]
LGTM but you can do one more step in the clean-up and remove the `coq_macos_version` variable which is dead code now.

**Comment by Zimmi48** [comment]
@gares @MSoegtropIMC Unfortunately, I don't think we have any test of the platform scripts on macOS in our CI yet. Can you confirm that none of this code is useful to generate the DMG bundle of the platform?

**Comment by MSoegtropIMC** [comment]
I can confirm the the Coq Platform DMG generation runs after building Coq via opam without using Coq makefiles in any way. It works in a similar way as the Windows installer by asking opam for installed files.

**Comment by gares** [comment]
We could still have a job (as we do for windows) which builds a minimal platform on osx.

**Comment by gares** [comment]
@MSoegtropIMC don't we still use the plist.template file from the Coq sources? This pr removes it.

**Comment by MSoegtropIMC** [comment]
I was about to move it over to Coq Platform since it is in a different folder in 8.12 and 8.13.

I hope we can agree that this file is mostly generated by the XCode plist editor with negligible creative content, so that copyright does not apply.

**Comment by Zimmi48** [comment]
Will merge when CI passes.

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**